app/src/views/CountryNsOverviewContextAndStructure/NationalSocietyLocalUnits/LocalUnitDeleteModal/index.tsx

```diff
@@ -0,0 +1,198 @@
+import {
+    useCallback,
+    useMemo,
+} from 'react';
+import {
+    Button,
+    Modal,
+    RadioInput,
+    TextArea,
+} from '@ifrc-go/ui';
+import { useTranslation } from '@ifrc-go/ui/hooks';
+import { resolveToString } from '@ifrc-go/ui/utils';
+import {
+    createSubmitHandler,
+    getErrorObject,
+    ObjectSchema,
+    PartialForm,
+    requiredStringCondition,
+    useForm,
+} from '@togglecorp/toggle-form';
+
+import type { GlobalEnums } from '#contexts/domain';
+import useAuth from '#hooks/domain/useAuth';
+import useGlobalEnums from '#hooks/domain/useGlobalEnums';
+import useAlert from '#hooks/useAlert';
+import {
+    GoApiBody,
+    useLazyRequest,
+} from '#utils/restRequest';
+
+import i18n from './i18n.json';
+import styles from './styles.module.css';
+
+type DeprecateReason = NonNullable<GlobalEnums['local_units_deprecate_reason']>[number];
+
+type LocalUnitDeprecateBody = GoApiBody<'/api/v2/local-units/{id}/deprecate/', 'POST'>;
+
+type DeprecateFormType = PartialForm<LocalUnitDeprecateBody>;
+type FormSchema = ObjectSchema<DeprecateFormType>;
+type FormSchemaFields = ReturnType<FormSchema['fields']>;
+
+const deprecateReasonKeySelector = (
+    item: DeprecateReason,
+) => item.key;
+
+const deprecateReasonLabelSelector = (
+    item: DeprecateReason,
+) => item.value;
+
+const defaultFormValue: DeprecateFormType = {
+};
+
+const schema: FormSchema = {
+    fields: (): FormSchemaFields => ({
+        deprecated_reason: {
+            required: true,
+        },
+        deprecated_reason_overview: {
+            required: true,
+            requiredValidation: requiredStringCondition,
+        },
+    }),
+};
+
+interface Props {
+    localUnitId: number;
+    onDeleteActionSuccess?: () => void;
+    onClose: () => void;
+    localUnitName: string;
+    disabled?: boolean;
+}
+
+function LocalUnitDeleteModal(props: Props) {
+    const strings = useTranslation(i18n);
+    const {
+        localUnitId,
+        localUnitName,
+        onDeleteActionSuccess,
+        onClose,
+        disabled,
+    } = props;
+
+    const {
+        local_units_deprecate_reason: deprecateReasonOptions,
+    } = useGlobalEnums();
+
+    const {
+        value,
+        error: formError,
+        setFieldValue,
+        setError,
+        validate,
+    } = useForm(schema, { value: defaultFormValue });
+
+    const error = useMemo(
+        () => getErrorObject(formError),
+        [formError],
+    );
+
+    const { isAuthenticated } = useAuth();
+
+    const hasDeletePermission = isAuthenticated;
```

Comment:
This is still and issue. The way it was renamed was not the isssue.
The issue is regarding the logic.
Do we not have permissions defined on the server on what a user can perform?
Not all authenticated user should have delete permission.
example: we have guest users who can login but cannot perform any operation.